src/main/java/io/gingersnapproject/k8s/CacheRuleInformer.java

```diff
@@ -0,0 +1,224 @@
+            processConfigMapAndSend(changed, (name, rule) -> {
+               log.info("calling DynamicRuleManagement.updateRule({},{}", name, rule.toString());
+               throw new UnsupportedOperationException("Rules cannot be updated: "+changed.toString());
+            // this.drm.updateRule(name,new EagerCacheRuleSpecAdapter(rule));
+         });
+      }
+         processConfigMapAndSend(removed, (name, rule) -> {
+            log.info("calling DynamicRuleManagement.removeRule({},{}", name, rule.toString());
+            this.drm.removeRule(name);
+         });
+         processConfigMapAndSend(added, (name, rule) -> {
+            log.info("calling DynamicRuleManagement.addRule({},{}", name, rule.toString());
+            this.drm.addRule(name,new EagerCacheRuleSpecAdapter(rule));
+         });
+      }
+
+      @Override
+      public void onDelete(ConfigMap obj, boolean deletedFinalStateUnknown) {
+         processConfigMapAndSend(obj.getData().entrySet(), (name, rule) -> {
+            log.info("calling DynamicRuleManagement.removeRule({},{}", name, rule.toString());
+            this.drm.removeRule(name);
+         });
+      }
+   }
+}
+
+// Used only by CacheRuleInformer
+class EagerCacheRuleSpecAdapter implements Rule {
+   private final class ConnectorImplementation implements Connector {
+      private final String[] schemaTable;
+
+      private ConnectorImplementation(String[] schemaTable) {
+         this.schemaTable = schemaTable;
+      }
+
+      @Override
+      public String schema() {
+         return schemaTable[0];
+      }
+
+      @Override
+      public String table() {
+         return schemaTable[1];
+      }
+   }
+
+   private EagerCacheRuleSpec eagerRule;
+
+   public EagerCacheRuleSpecAdapter(EagerCacheRuleSpec eagerRule) {
+      this.eagerRule = eagerRule;
+   }
+
+   @Override
+   public Connector connector() {
+      String[] schemaTable = eagerRule.getTableName().split("\\.");
+      return switch(schemaTable.length) {
+         case 1 -> new Connector() {
+            @Override
+            public String schema() {
+               return "";
+            }
+
+            @Override
+            public String table() {
+               return schemaTable[0];
+            }
+         };
+         case 2 -> new ConnectorImplementation(schemaTable);
+         default ->
+            throw new IllegalArgumentException("Unsupported schema format (must be table or schema.table)");
+      };
+   }
+
+   @Override
+   public KeyFormat keyType() {
+      return eagerRule.getKey().getFormat();
+   }
+
+   @Override
+   public String plainSeparator() {
+      return eagerRule.getKey().getKeySeparator();
+   }
+
+   @Override
+   public Optional<List<String>> keyColumns() {
+      return Optional.ofNullable(eagerRule.getKey().getKeyColumnsList());
+   }
+
+   @Override
+   public Optional<List<String>> columns() {
+      return Optional.ofNullable(eagerRule.getValue().getValueColumnsList());
```

Comment:
For this and `keyColumns`, Intellij is complaining that we're passing a non-null arg to `ofNullable` and that we should use `.of` instead.